core/lib/health_check/src/metrics.rs

```diff
@@ -0,0 +1,43 @@
+//! Metrics for health checks.
+
+use std::time::Duration;
+
+use vise::{Buckets, EncodeLabelSet, EncodeLabelValue, Family, Histogram, Metrics, Unit};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue)]
+#[metrics(rename_all = "snake_case")]
+pub(crate) enum CheckResult {
+    Slow,
+    TimedOut,
+    Dropped,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelSet)]
+struct AbnormalCheckLabels {
+    component: &'static str,
+    result: CheckResult,
+}
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "healthcheck")]
+pub(crate) struct HealthMetrics {
+    /// Latency for abnormal checks. Includes slow, dropped and timed out checks (distinguished by the "result" label);
+    /// skips normal checks.
```

Comment:
Hmm why do we want to skip normal checks? Because there will a lot of very fast data points? Outliers will still be visible on the heatmap..

Reply:
Yes, the issue is with the number of data points. IIUC, this is why we have time series for the corresponding built-in K8s metrics disabled (e.g., no `prober_probe_duration_seconds_bucket` time series, while `prober_probe_duration_seconds_sum` and `*_count` time series are present).

Reply:
this is not directly related to this PR, so can be tackled separately. The PR I'd like to merge ASAP